app/assets/javascripts/float-kit/addon/components/d-menu.gjs

```diff
@@ -104,7 +106,11 @@ export default class DMenu extends Component {
 
     return (
       this.args.triggerComponent ||
-      curryComponent(DButton, baseArguments, getOwner(this))
+      curryComponent(
+        this.args.triggerTag ? element(this.args.triggerTag) : DButton,
```

Comment:
@davidtaylorhq I know you already told me in the past that you would prefer if we could avoid this, but I couldnt make my API work without this. This is what I tried before:


```
const CustomTriggerComponent = <template>
{{#each this.args.selection as |selection|}}
  {{yield selection to="selection"}}
 {{/each}} 
</template>
```

And on the DMenu `@triggerComponent=(component CustomTriggerComponent selection=this.selection)`

The yield wouldn't work in this case due to the nesting I imagine.

Reply:
Right, if you want the `{{yield}}` to target a named block of DMultiSelect, then it needs to be in DMultiSelect's template. But I don't see why that means we have to add a new `@triggerTag` argument?

Couldn't DMultiSelect do something like this?

```
<DMenu @trigger={{element "div"}}>
  <:trigger>
    ...
    {{yield selection to="selection"}}
    ...
  </:trigger>
</DMenu>
```

Reply:
Hmm right that might just work will try indeed